src/MDMAnimationTraits.h

```diff
@@ -0,0 +1,114 @@
+/*
+ Copyright 2017-present The Material Motion Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <CoreGraphics/CoreGraphics.h>
+#import <Foundation/Foundation.h>
+
+#import "MDMRepetitionTraits.h"
+#import "MDMTimingCurve.h"
+
+/**
+ A generic representation of animation traits.
+ */
+@interface MDMAnimationTraits: NSObject
+
+/**
+ Initializes the instance with the provided duration and default iOS ease in/out cubic bezier.
+
+ @param duration The animation will occur over this length of time, in seconds.
+ */
+- (nonnull instancetype)initWithDuration:(NSTimeInterval)duration;
+
+/**
+ Initializes the instance with the provided duration, delay, and default iOS ease in/out cubic
+ bezier.
+
+ @param delay The amount of time, in seconds, to wait before starting the animation.
+ @param duration The animation will occur over this length of time, in seconds, after the delay time
+ has passed.
+ */
+- (nonnull instancetype)initWithDelay:(NSTimeInterval)delay duration:(NSTimeInterval)duration;
```

Comment:
Nit: This should probably be "initWithDuration:delay:" to match the convenience constructor above. Same for the two below.

Reply:
Have flip-flopped on this. Conceptually the delay happens before the duration, so I was attempting to make it easier to scan and understand blocks of animation traits. For example:

```objc
- (MDMAnimationTraits *)chipWidth {
  return [[MDMAnimationTraits alloc] initWithDelay:0.000 duration:0.285 timingCurve:StandardTimingCurve()];
}

- (MDMAnimationTraits *)chipHeight {
  return [[MDMAnimationTraits alloc] initWithDelay:0.015 duration:0.360 timingCurve:StandardTimingCurve()];
}

- (MDMAnimationTraits *)chipY {
  return [[MDMAnimationTraits alloc] initWithDelay:0.015 duration:0.360 timingCurve:StandardTimingCurve()];
}

- (MDMAnimationTraits *)chipContentOpacity {
  return [[MDMAnimationTraits alloc] initWithDelay:0.000 duration:0.075 timingCurve:LinearTimingCurve()];
}

- (MDMAnimationTraits *)headerContentOpacity {
  return [[MDMAnimationTraits alloc] initWithDelay:0.075 duration:0.150 timingCurve:LinearTimingCurve()];
}

- (MDMAnimationTraits *)navigationBarY {
  return [[MDMAnimationTraits alloc] initWithDelay:0.015 duration:0.360 timingCurve:StandardTimingCurve()];
}
```

Reply:
That makes sense. Not a blocking issue, just jumped out at me when I was reading.